Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs

```diff
@@ -0,0 +1,210 @@
+﻿using Xunit;
+using Optiq.DatePicker;
+
+namespace Optiq.DatePicker.Tests;
+
+public class DatePickerUnitTests
+{
+    [Fact]
+    public void TestMinimumDate()
+    {
+        Optiq.DatePicker.DatePicker picker = new Optiq.DatePicker.DatePicker();
+
+        picker.MinimumDate = new DateTime(1950, 1, 1);
+
+        Assert.Equal(new DateTime(1950, 1, 1), picker.MinimumDate);
+
+        
+        Assert.Throws<ArgumentException>(() => picker.MinimumDate = new DateTime(2200, 1, 1));
+    }
+
+
+    [Fact]
+    public void TestNullableDate()
+    {
+        Optiq.DatePicker.DatePicker picker = new Optiq.DatePicker.DatePicker();
+        if (picker.Date == null)
+        {
+            Assert.Null(picker.Date);
+        }
+
+    }
+
+    [Fact]
+    public void TestMaximumDate()
+    {
+        Optiq.DatePicker.DatePicker picker = new Optiq.DatePicker.DatePicker();
+
+        picker.MaximumDate = new DateTime(2050, 1, 1);
+
+        Assert.Equal(new DateTime(2050, 1, 1), picker.MaximumDate);
+
+        Assert.Throws<ArgumentException>(() => picker.MaximumDate = new DateTime(1800, 1, 1));
+    }
+
+    [Fact]
+    public void TestMaximumDateClamping()
+    {
+        Optiq.DatePicker.DatePicker picker = new Optiq.DatePicker.DatePicker();
+
+        picker.Date = new DateTime(2050, 1, 1);
+
+        Assert.Equal(new DateTime(2050, 1, 1), picker.Date);
+
+        bool dateChanged = false;
+        bool maximumDateChanged = false;
+        picker.PropertyChanged += (sender, e) =>
+        {
+            switch (e.PropertyName)
+            {
+                case "MaximumDate":
+                    maximumDateChanged = true;
+                    break;
+                case "Date":
+                    dateChanged = true;
+                    Assert.False(maximumDateChanged);
+                    break;
+            }
+        };
+
+        var newDate = new DateTime(2000, 1, 1);
+        picker.MaximumDate = newDate;
+
+        Assert.True(maximumDateChanged);
+        Assert.True(dateChanged);
```

Comment:
also I don't get it, is you change the maximum date, why should the date also change?